src/main/java/com/cloudbees/plugins/credentials/casc/SystemCredentialsProviderConfigurator.java

```diff
@@ -64,17 +75,29 @@ protected SystemCredentialsProvider instance(Mapping mapping, ConfigurationConte
     public Set<Attribute<SystemCredentialsProvider, ?>> describe() {
         return Collections.singleton(
             new MultivaluedAttribute<SystemCredentialsProvider, DomainCredentials>("domainCredentials", DomainCredentials.class)
-                .setter( (target, value) -> target.setDomainCredentialsMap(DomainCredentials.asMap(value)))
+                .setter((target, value) -> {
+                    String strategy = Util.fixEmptyAndTrim(
+                            System.getProperty("casc.credentials.merge.strategy",
+                            System.getenv("CASC_CREDENTIALS_MERGE_STRATEGY")
+                    ));
```

Comment:
rather than put the strategy in an environment variable - is it not possible to manage it purely as code in the yaml file?

Reply:
I'd go even further and define the default strategy value as `replace` to make it explicit.  Example yaml might look like:

```
- MergeStrategies
    - CascCredentials: merge
````